src/lib/pipeline.ml

```diff
@@ -587,12 +593,47 @@ module Compiler = struct
     in
     compiler.wrap_bam_node pipeline bam_node
 
-  let compile_variant_caller_step ~compiler (pipeline: vcf pipeline) =
+  let rec compile_variant_caller_step ~compiler (pipeline: vcf pipeline) =
     let {processors ; reference_build; work_dir; machine ;} = compiler in
     let result_prefix = work_dir // to_file_prefix pipeline in
     dbg "Result_Prefix: %S" result_prefix;
     let vcf_node =
       match pipeline with
+      | Somatic_variant_caller
+          (som_vc,
+           Bam_pair (
+             Gatk_bqsr (n_bqsr_config, Gatk_indel_realigner (n_gir_conf, n_bam))
+             , 
+             Gatk_bqsr (t_bqsr_config, Gatk_indel_realigner (t_gir_conf, t_bam))
+           )) 
+        when has_option compiler `Multi_sample_indel_realignment
+          && n_gir_conf = t_gir_conf ->
```

Comment:
might be good to throw if the configurations are different but `Multi_sample_indel_realignment is specified; that seems like a clear user error